### ISSUE TITLE
open a public sheet by using an API key instead of an oauth

### ISSUE THREAD

**Issue by gitagogaming** [burnash/gspread#1096]
- _**Is your feature request related to a problem? Please describe.**_
   - Accessing a sheet which is designated as public data with Oauth causes rate limit due to multiple users. 
Using an API key directly would allow more updates per second to get the sheet values

---
- **_Describe the solution you'd like_**
    - A new function to pull a sheets values similar to ```open_by_key(sheet_id)``` but allowing an option to pass an API key instead of the oauth tokens
---
- _**Describe alternatives you've considered**_
   - as far as alternatives I've considered.. would be another library but I would like to avoid that 
---
- _**Additional context**_
    - Add any other context or screenshots about the feature request here. [Link to other](https://github.com/xflr6/gsheets/blob/24e99f67e5f6bc49facc85f5db305caa63ccbdef/gsheets/api.py#L38)

**Comment by lavigne958** [comment]
Hi, thank you for raising this issue. This is interesting, I did not know about this _developer key_ / _API key_. I will have a look at it and see if we can add it to gspread.

If this raises the API rate limits, then it would be awesome to have it.

**Comment by lavigne958** [comment]
Hi yes it has been, i need time to find a clean solution that integrates with the current system. 

Currently gspread is built with mandatory auth system as entry point.

I have ideas I'll test them and come back with a solution

**Comment by lavigne958** [comment]
Hi I found a way to use the API_Key to authenticate against google APIs and read publicly shared spreadsheets.

I managed to read the content of a spreadsheet with it.
I faced some API rate limiting, quotas. I received the error message with a quota exceeded of 60 read requests / seconds.

This could be introduced but it won't solve the quota issue.

I will still plan to introduce that new way to open sheets.
